Kernel/Clause.cpp

```diff
@@ -201,6 +205,9 @@ void Clause::destroy()
   static Stack<Clause*> toDestroy(32);
   Clause* cl = this;
   for(;;) {
+    if ((env.options->proofExtra()==Options::ProofExtra::FULL) && env.proofExtra) {
```

Comment:
Indeed.

Reply:
Could I just curiously inquire what kind of bug? 

You mean `proofExtra` storing info for a deleted clause? That means a memory leak? Or something more dangerous?

Reply:
Since I wrote this code already quite some time ago, I don't remember exactly what the bug was - but I think it was about storing info for a deleted clause, then allocating new clause in the same memory slot, and hence having wrong info for the new clause.

Reply:
Already does! https://github.com/vprover/vampire/blob/master/Lib/Hash.hpp#L145 :-)